### PR TITLE
Adding Nested Function Documentation

### PR DESCRIPTION
**Pull request by forestmvey**
### Description
Added documentation section in SQL for nested function added to V2 engine in  [PR-1490](https://github.com/opensearch-project/sql/pull/1490) and [PR-1657](https://github.com/opensearch-project/sql/pull/1657). The added features covers the `nested` function used in SQL.

### PDF
[Nested Function - OpenSearch documentation.pdf](https://github.com/Bit-Quill/documentation-website/files/11737164/Nested.Function.-.OpenSearch.documentation.pdf)

 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Yury-Fridlyand — CHANGES_REQUESTED** [review]
This is documentation for entire plugin, not for V2. Mention that in V2 it is supported in `SELECT` clause only and reference [this paragraph](https://opensearch.org/docs/latest/search-plugins/sql/limitation/#query-processing-engines).
But you have to share examples and syntax for V1 too and for WHERE clause too.
Fortunately, this won't change with your implementation in V2.

Please also add entire web page (not MD) as PDF to the PR, so we can see how it looks like.

File: _search-plugins/sql/sql/functions.md [hunk 1]
```diff
@@ -225,3 +225,63 @@ The results contain documents that match the wildcard expression:
...
+## Nested query
+
+The nested function is used in the `SELECT` clause to unnest nested object type collections. The nested collection is flattened and a cartesian product is returned when querying against two nested objects.
+
+### Syntax
+
+The `field_expression` parameter is required, and the `path_expression` parameter is optional. Dot notation is used to show nesting level for `field_expression` and `path_expression` parameters. For example `nestedObj.innerFieldName` denotes a field nested one level. If the user does not provide the `path_expression` parameter it will be generated dynamically. For example the field `user.office.cubicle` would dynamically generate the path `user.office`.
+
+```sql
+nested(field_expression | field_expression, path_expression)
```
Comment (matthewryanwells) [inline _search-plugins/sql/sql/functions.md]: Would it be worth putting the example before the description so you can see the layout of the expression before the explanation?

**Review by GumpacG — APPROVED** [review]

File: _search-plugins/sql/sql/functions.md [hunk 2]
```diff
@@ -225,3 +225,63 @@ The results contain documents that match the wildcard expression:
 | account_number | address
 :--- | :---
 1 | 880 Holmes Lane
+
+## Nested query
+
+The nested function is used in the `SELECT` clause to unnest nested object type collections. The nested collection is flattened and a cartesian product is returned when querying against two nested objects.
```
Comment (acarbonetto) [inline _search-plugins/sql/sql/functions.md]: When used in the SELECT clause?

Comment (acarbonetto) [reply _search-plugins/sql/sql/functions.md]: This document doesn't mention anything about the WHERE clause.  Maybe we can add that too?

**Review by acarbonetto — APPROVED** [review]

File: _search-plugins/sql/sql/functions.md (hunk 1 above)
Comment (forestmvey) [reply _search-plugins/sql/sql/functions.md]: Doesn't tend to be the standard way that the examples are presented on the documentation. I'd like to conform with the other pages.

File: _search-plugins/sql/sql/nested.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+nav_order: 13
+redirect_from:
+- /search-plugins/sql/nested/
+---
+
+# Nested
+
+## Nested In SELECT Clause
+
+The nested function is used in the `SELECT` clause to unnest nested object type collections. The nested collection is flattened and a cartesian product is returned when querying against two or more nested collections. The nested function is supported in the V2 engine in the `SELECT` and `WHERE` clauses, see [query-processing-engines](https://opensearch.org/docs/latest/search-plugins/sql/limitation/#query-processing-engines) for more details.
```
Comment (Yury-Fridlyand) [inline _search-plugins/sql/sql/nested.md]: I feel that phrase
> The nested function is supported

is not suitable here. Maybe you need to move it outside of this paragraph, for example to a separate section on the very bottom called 'limitations'.

File: _search-plugins/sql/sql/nested.md [hunk 4]
```diff
@@ -0,0 +1,211 @@
+---
+layout: default
+title: Nested Function
+parent: SQL
+grand_parent: SQL and PPL
+nav_order: 13
```
Comment (Yury-Fridlyand) [inline _search-plugins/sql/sql/nested.md:6]: toc = table of content (generated automatically)
```suggestion
nav_order: 13
has_toc: true
```

**Review by Yury-Fridlyand — APPROVED** [review]

File: _search-plugins/sql/sql/nested.md [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+      { "info": "letter2" }
+    ]
+  }
+}
+```
+
+The results contain documents that match the nested query:
+
+| nested(message.info) |
+:----------------------|
```
Comment (MitchellGale) [inline _search-plugins/sql/sql/nested.md:127]: Is this the correct table format and not:
```
| nested(message.info) |
| -------------------- |
| letter2              |

**Pull request merged**